### PR TITLE
Updates the library to the latest version

### LINKED ISSUE #4346: Update import paths for github.com/devfile/api to github.com/devfile/api/v2
We've recently updated the module path for the [github.com/devfile/api](https://github.com/devfile/api) Go module to now be `github.com/devfile/api/v2`, to enable us to release tagged versions of the Go module that match the Devfile schema's version (e.g. module version v2.0.0 for a 2.0.0 schema version). Some discussion on why this was needed can be found [here](https://github.com/golang/go/wiki/Modules#releasing-modules-v2-or-higher).

This doesn't need to be done immediately, but once a new version of the devfile api is pulled into odo, the import paths for the Devfile API package that odo uses will need to be updated to change instances of `import github.com/devfile/api/pkg...` in odo to `import github.com/devfile/api/v2/pkg...`. (e.g. in https://github.com/openshift/odo/blob/adaa1b44e8de5ab0aa0d357294174f8efb8bdd24/pkg/odo/cli/storage/create.go#L7 and https://github.com/openshift/odo/blob/526ce0af8ce506ea4709595a5a9abb160ea3abe1/pkg/url/status_test.go#L8, among other instances)

### PR DESCRIPTION
**Pull request by mik-dass**
**What type of PR is this?**

/kind code-refactoring

**What does does this PR do / why we need it**:

Updates the library (https://github.com/devfile/library) to the latest version 

**Which issue(s) this PR fixes**:

Fixes #4346 

**PR acceptance criteria**:

- [X] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

Tests should pass.

### CODE REVIEW

**Review by dharmit — APPROVED** [review]
/lgtm

**Comment by dharmit** [comment]
@mik-dass I've lgtm'd since it's mostly unit test changes and they've already passed.

**Comment by dharmit** [comment]
All integration failures on PSI are happening at `odo create`.

File: tests/examples/source/devfiles/nodejs/devfile-with-no-default.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      mountSources: true
+commands:
+  - id: firstbuild
+    exec:
+      component: runtime
+      commandLine: npm install
+      workingDir: ${PROJECTS_ROOT}
+      group:
+        kind: build
+        isDefault: true
```
Comment (dharmit) [inline tests/examples/source/devfiles/nodejs/devfile-with-no-default.yaml:26]: File name here is `devfile-with-no-default.yaml` but there's a `isDefault: true` here. Or is this devfile about no default `run` command?

**Review by dharmit — COMMENTED** [review]
Just one question

File: tests/examples/source/devfiles/nodejs/devfile-with-no-default.yaml (hunk 1 above)
Comment (mik-dass) [reply tests/examples/source/devfiles/nodejs/devfile-with-no-default.yaml:26]: Yes, it is about no default run command.

**Comment by mik-dass** [comment]
```
Get https://image-registry.openshift-image-registry.svc:5000/v2/: dial tcp -IP REDACTED-:5000: i/o timeout
[ssh:Fedora 32] [odo] I0222 10:43:27.920748 2879502 buildConfigs.go:137] Status of nodejs-myapp-1 build is Running
[ssh:Fedora 32] [odo] I0222 10:43:27.921262 2879502 buildConfigs.go:137] Status of nodejs-myapp-1 build is Running
[ssh:Fedora 32] [odo] I0222 10:43:28.047410 2879502 buildConfigs.go:137] Status of nodejs-myapp-1 build is Failed
[ssh:Fedora 32] [odo]  ✗  Waiting for build to finish [3m]
[ssh:Fedora 32] [odo]  ✗  Failed to create component with name nodejs. 
```
/retest

**Pull request merged**